### PR TITLE
Expo 51

### PR DESCRIPTION
**Pull request by Koalk**
Add support for Expo 51

---------------------

Self Review:

* [x] Appropriate test coverage
* [x] Relevant Documentation updated

Smoke Tests:

* [x] Example app launches on iOS and Android

### CODE REVIEW

**Comment by Koalk** [comment]
I think we might be getting [this](https://github.com/expo/expo/issues/29190) which blocks android from building

**Comment by ball-hayden** [comment]
We probably need the Gradle changes from https://github.com/PlayerData/react-native-mcu-manager/commit/5fa1c78ba64ca535cb0692351d248c3e46324c42#diff-197b190e4a3512994d2cebed8aff5479ff88e136b8cc7a4b148ec9c3945bd65a

File: example/pnpm-workspace.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+packages:
```
Comment (ball-hayden) [inline example/pnpm-workspace.yaml:1]: We shouldn't need a workspace definition for the example app - it's part of the parent workspace.

File: pnpm-workspace.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+packages:
+  # include packages in subfolders (e.g. apps/ and packages/)
+  - "ios/**"
+  - 'android/**'
+  - 'src/**'
```
Comment (ball-hayden) [inline pnpm-workspace.yaml:5]: These aren't really separate packages.

We've got one package, then the example app. Not sure how we deal with that when the example app is nested inside the root package.

**Review by saty9 — APPROVED** [review]

**Pull request merged**